### PR TITLE
Build prefab docker for prover

### PR DESCRIPTION
**Pull request by Ancient123**
By default includes the prover 1 config
Has the capability to pull in configs from aws secrets manager

### CODE REVIEW

File: .github/workflows/prover-docker.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate docker metadata
+        uses: docker/metadata-action@v4
+        id: prover-metadata
+        with:
+          images: ghcr.io/${{ github.repository_owner }}/espresso-polygon-zkevm-demo/zkevm-prover
+
+      - name: Build and push geth-with-contracts docker
```
Comment (sveitser) [inline .github/workflows/prover-docker.yml:43]:
```suggestion
      - name: Build and push prover docker
```

File: .github/workflows/prover-docker.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: Postgres state-db docker
```
Comment (sveitser) [inline .github/workflows/prover-docker.yml:1]:
```suggestion
name: zkevm-prover docker
```

File: docker/scripts/zkevm-prover.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -eEu -o pipefail
+
+if [[ -v CONFIG_SECRET_NAME ]]; then
+  echo "Loading config file from AWS secrets manager"
+   aws secretsmanager  get-secret-value --secret-id ${CONFIG_SECRET_NAME} --query SecretString --output text | tee /app/config.json
```
Comment (jbearer) [inline docker/scripts/zkevm-prover.sh:6]: Should we be `tee`ing this if it's a secret?

Comment (Ancient123) [reply docker/scripts/zkevm-prover.sh:6]: We should definitely not be echoing that into stdout. Let me fix that..

**Pull request merged**